tests/test_splitter.py

```diff
@@ -0,0 +1,25 @@
+import os
+import pytest
+from pathlib import Path
+
+from fastembed.embedding import DefaultEmbedding
+
+@pytest.mark.parametrize(["chunk_size", "chunk_overlap"], [[500, 50], [1000, 100]])
+def test_embedding(chunk_size: int, chunk_overlap: int):
+    is_ubuntu_ci = os.getenv("IS_UBUNTU_CI")
+
+    for model_desc in DefaultEmbedding.list_supported_models():
+        if is_ubuntu_ci == "false" and model_desc["size_in_GB"] > 1:
+            continue
+
+        p = Path(__file__).with_name("state_of_the_union.txt")
+        with open(p) as f:
+            text = f.read()
+            embedding = DefaultEmbedding(
+                model_name=model_desc["model"],
+                chunk_size=chunk_size,
+                chunk_overlap=chunk_overlap,
+            )
```

Comment:
The `chunk_size` should be in tokens and not characters. And if the user passes a value larger than the Embedding model token sequence length, we should raise a ValueError.

Reply:
The chunk size is calculated based on the tokenizers encoding.

https://github.com/qdrant/fastembed/blob/text-splitter/fastembed%2Fsplitter.py#L98

Reply:
Yes. I've added a comment as well. Please read the max tokenzier lenght from the tokenizer object. Not the config file.

Reply:
Never mind, I've added this to the validator. Please add a test for this i.e. that the error is indeed raised — and then mark this conversation as resolved.